src/Components/Notifications/useNotificationsContext.tsx

```diff
@@ -0,0 +1,49 @@
+import { useRouter } from "found"
+import { createContext, FC, useState, useContext, useEffect } from "react"
+
+export type State = {
+  currentNotificationId: string | null
+}
+
+export const DEFAULT_STATE: State = {
+  currentNotificationId: null,
+}
+
+const NotificationsContext = createContext<{
+  state: State
+  setCurrentNotificationId: (id: string) => void
+}>({
+  state: DEFAULT_STATE,
+  setCurrentNotificationId: () => {},
+})
+
+interface NotificationsContextProviderProps {
+  id?: string | null
+}
+
+export const NotificationsContextProvider: FC<NotificationsContextProviderProps> = ({
+  children,
+  id = null,
+}) => {
+  const { match } = useRouter()
+  const [currentNotificationId, setCurrentNotificationId] = useState(id)
+
+  useEffect(() => {
+    setCurrentNotificationId(match.params.notificationId)
+  }, [match.params.notificationId, setCurrentNotificationId])
+
+  return (
+    <NotificationsContext.Provider
+      value={{
+        state: { currentNotificationId },
+        setCurrentNotificationId,
+      }}
```

Comment:
#minor, but if you assign the contents of `value` to a variable, it leads to more stable renders:

```tsx
const contextProps = {
  ...
}

<NotificationsContext.Provider value={contextProps}>.
```

Reply:
@damassi do you have a reference for this? Intuitively, that doesn't make sense to me.

Reply:
I'm looking for it! But i know what you mean. It seems like a `useMemo` would be required

Reply:
Yeah, it looks like useMemo is how it is. But i could swear by experience that I stumbled upon this in the past, that for some reason the react render lifecycle would get funny on fully anonymous objects passed to contet, even tho they seem the same as passing an object. Anyways, nbd.